.eslintrc.js

```diff
@@ -6,38 +23,116 @@ module.exports = {
                     },
                     {
-                        name: 'react-native',
-                        importNames: ['TouchableOpacity', 'TouchableWithoutFeedback', 'TouchableNativeFeedback', 'TouchableHighlight'],
-                        message: 'Please use PressableWithFeedback and/or PressableWithoutFeedback from src/components/Pressable instead',
+                        selector: 'function',
+                        format: ['camelCase', 'PascalCase'],
                     },
                     {
-                        name: 'react-native',
-                        importNames: ['StatusBar'],
-                        message: 'Please use StatusBar from src/libs/StatusBar instead',
+                        selector: ['typeLike', 'enumMember'],
+                        format: ['PascalCase'],
+                    },
+                    {
+                        selector: ['parameter', 'method'],
+                        format: ['camelCase'],
+                    },
+                ],
+                '@typescript-eslint/array-type': ['error', {default: 'array-simple'}],
+                '@typescript-eslint/no-unsafe-enum-comparison': 'error',
+                '@typescript-eslint/prefer-enum-initializers': 'error',
+                '@typescript-eslint/no-var-requires': 'off',
+                '@typescript-eslint/no-non-null-assertion': 'error',
+                '@typescript-eslint/no-explicit-any': 'error',
+                '@typescript-eslint/switch-exhaustiveness-check': 'error',
+                '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
+                'es/no-nullish-coalescing-operators': 'off',
+                'es/no-optional-chaining': 'off',
+                'valid-jsdoc': 'off',
+                'jsdoc/no-types': 'error',
+                'import/no-extraneous-dependencies': 'off',
+                'rulesdir/onyx-props-must-have-default': 'off',
+                'no-restricted-syntax': [
+                    'error',
+                    {
+                        selector: 'TSEnumDeclaration',
+                        message: "Please don't declare enums, use union types instead.",
+                    },
+                ],
+                'no-restricted-imports': [
+                    'error',
+                    {
+                        paths: [
+                            ...globalRestrictedImport,
+                            {
+                                name: 'lodash/get',
+                                message: 'Please use optional chaining and nullish coalescing instead.',
+                            },
+                            {
+                                name: 'lodash',
+                                importNames: ['get'],
+                                message: 'Please use optional chaining and nullish coalescing instead.',
+                            },
+                            {
+                                name: 'underscore',
```

Comment:
Let's ban `underscore` itself per[ this discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1687477811950459)/

```js
{
  name: 'underscore',
  message: 'Please use corresponding method from lodash instead',
```

Reply:
Banned both, `underscore` and `object` type.